starky/src/lookup.rs

```diff
@@ -579,11 +579,6 @@ pub(crate) fn lookup_helper_columns<F: Field>(
     challenge: F,
     constraint_degree: usize,
 ) -> Vec<PolynomialValues<F>> {
-    assert!(
-        constraint_degree <= 3,
-        "TODO: Allow other constraint degrees."
-    );
```

Comment:
I think we'd rather keep the assert, but with the updated condition instead.

Reply:
I moved the assert to `prove_with_commitment` since you don't know the `rate_bits` there (and is also weird to check the constraint degree in the lookups)